index.js

```diff
@@ -54,7 +54,20 @@ async function main() {
     const restored = await cache.restoreCache(cachePaths, cacheKey);
     const ret = await exec.exec('pre-commit', args, {ignoreReturnCode: push});
     if (!restored) {
-        await cache.saveCache(cachePaths, cacheKey);
+        try {
+            await cache.saveCache(cachePaths, cacheKey);
+        } catch (e) {
+            core.warning(
+                `There was an error saving the pre-commit environments to cache:
+
+                ${e.message ? e.message : e}
+
+                This only has performance implications and won't change the result of your pre-commit tests.
+                If this problem persists on your default branch, you can try to fix it by editing your '.pre-commit-config.yaml'.
+                I.e. run 'pre-commit autoupdate' or simply add a blank line.
```

Comment:
I think you actually mean `e.g.` -- though I try and avoid both of them because they're easy to confuse

Reply:
Oh yeah, I meant `e.g.` .... and ofc confused them. 😓 
Would replacing it with `For example` be fine?

Reply:
Replaced it with `For example try to`, hope that's fine.